components/tools/OmeroJava/test/integration/LightAdminRolesTest.java

```diff
@@ -279,203 +146,200 @@ private EventContext loginNewAdmin(boolean isAdmin, List <String> permissions) t
+        /* add a file attachment with original file to the imported image.*/
+        final ImageAnnotationLink fileAnnotationLink = linkImageAnnotation(image, mmFactory.createFileAnnotation());
+        /* link was saved in previous step with the whole graph, including fileAnnotation and original file */
+        final FileAnnotation fileAnnotation = (FileAnnotation) fileAnnotationLink.getChild();
+        final OriginalFile annotOriginalFile = fileAnnotation.getFile();
+        /* make a list of annotation objects in order to simplify checking of owner and group */
+        List<IObject> annotOriginalFileAnnotationTagAndLinks = new ArrayList<IObject>();
+        annotOriginalFileAnnotationTagAndLinks.addAll(Arrays.asList(annotOriginalFile, fileAnnotation, tagAnnotation,
+                tagAnnotationLink, fileAnnotationLink));
+        return annotOriginalFileAnnotationTagAndLinks;
     }
 
     /**
-     * Test that an ImporterAs can create new Project and Dataset
-     * and import data on behalf of another user solely with <tt>Sudo</tt> privilege
-     * into this Dataset. Further link the Dataset to the Project, check
-     * that the link belongs to the user (not to the ImporterAs).
-     * All workflows are tested here both when light admin is sudoing
-     * and when he/she is not sudoing, except for Link and Import (both tested
-     * only when sudoing, as the non-sudoing workflows are too complicated
-     * for those two actions and thus covered by separate tests).
+     * Light administrator (lightAdmin) tries to create new Project and Dataset and set normalUser as
+     * the owner of the Project and Dataset in a group where lightAdmin is not member (normalUser's group).
+     * lightAdmin tries the creation above both sudoing and not sudoing as normalUser
+     * and both having and not having <tt>WriteOwned<tt> privilege.
+     * The test finishes for test cases in which lightAdmin does not sudo after the creation attempt above.
+     * For cases in which lightAdmin does Sudo as normalUser:
+     * lightAdmin tries to import data on behalf of normalUser into the just created Dataset.
+     * lightAdmin, still sudoed as normalUser, tries then to link the Dataset to the Project.
+     * Non-sudoing workflows for import and linking are too complicated to be included in this test.
+     * Those two actions are covered by separate tests (testImporterAsNoSudoChownOnlyWorkflow,
+     * testImporterAsNoSudoChgrpChownWorkflow and testLinkNoSudo).
      * @param isSudoing if to test a success of workflows where Sudoed in
      * @param permWriteOwned if to test a user who has the <tt>WriteOwned</tt> privilege
      * @param groupPermissions if to test the effect of group permission level
      * @throws Exception unexpected
+     * @see <a href="graphical explanation">https://docs.google.com/presentation/d/1JX6b9pkPtG-3hZIGSp2bu4WVvNKn6GHqIKdfbbJkiw8/edit#slide=id.p4</a>
```

Comment:
https://github.com/openmicroscopy/openmicroscopy/blob/f81147e00/components/common/src/ome/api/IQuery.java#L262 may be a helpful example (complete with typo!). You can also paste (and don't commit!) guesses into other non-test OMERO classes, build the Javadoc then look at it.